code/modules/spells/passive/passive.dm

```diff
@@ -0,0 +1,10 @@
+//Passive spells
+//Cannot be normally cast, instead they rely on process() except for the no_clothes.dm spell
+
+/spell/passive
+    charge_type = Sp_PASSIVE
+    level_max = list(Sp_TOTAL = 0) //Passive spells have no use. For the love of God, do NOT give it Sp_SPEED, it will do nothing
+    charge_max = 0 //Redundancy
+
+/spell/passive/process()
```

Comment:
Where even is that proc defined on the first place ? I can't see it defined on `spell` level. Why does this compiles ?

Reply:
spell_code.dm
process() handles the cooldown/charge of spells
The gist is that passive spells do not have a cooldown and the parent instead returns nothing with the expectation that its children will have effects added to process()

Reply:
Well :
```

/spell/proc/take_charge(mob/user = user, var/skipcharge)
	if(!skipcharge)
		if(charge_type & Sp_RECHARGE)
			charge_counter = 0 //doesn't start recharging until the targets selecting ends
			src.process()
		if(charge_type & Sp_CHARGES)
			charge_counter-- //returns the charge if the targets selecting fails
		if(charge_type & Sp_HOLDVAR)
			if(special_var_holder)
				adjust_var(special_var_holder, holder_var_type, holder_var_amount)
			else
				adjust_var(user, holder_var_type, holder_var_amount)
		if(charge_type & Sp_GRADUAL)
			gradual_casting = TRUE
			charge_counter -= 1
			process()
```

Those are the only way `process` is called, if the spell is `Sp_GRADUAL` or `Sp_RECHARGE`.
Might want to make sure your `process` proc is actually called if your spell is going to be `Sp_PASSIVE`, because right now, your proc is a dud that doesn't ever get called.

Reply:
I will see what I can do